pkg/tests/cross-tests/rapid_tv_gen.go

```diff
@@ -0,0 +1,402 @@
+// Copyright 2016-2024, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Rapid generators for schema and value spaces.
+package crosstests
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-go/tftypes"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/contract"
+	"pgregory.net/rapid"
+)
+
+// Combines a TF value representing resource inputs with its schema. The value has to conform to the schema. The name
+// "tv" stands for a typed value.
+type tv struct {
+	schema   schema.Schema
+	typ      tftypes.Type
+	valueGen *rapid.Generator[tftypes.Value]
+}
+
+// Here "tb" stands for a typed block. Like [tv] but for non-nil/unknown blocks.
+type tb struct {
+	schemaMap map[string]*schema.Schema
+	typ       tftypes.Object
+	valueGen  *rapid.Generator[tftypes.Value]
+}
+
+type schemaT func(schema.Schema) schema.Schema
+
+type attrKind int
+
+const (
+	optionalAttr attrKind = iota + 1
```

Comment:
Why the `iota + 1`? If `0` is invalid then we should add a `invalidAttr = 0` to make that explicit.